### PR TITLE
Avoid unsupported probs with out of order wires

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This PR follows the same support for lightning PR https://github.com/PennyLaneAI/pennylane-lightning/pull/416 and prevents out of order wires being supplied directly to `probs`.

**Description of the Change:** As above.

**Benefits:** Prevents incorrect results being returned to user, and fails early instead.

**Possible Drawbacks:**

**Related GitHub Issues:** https://github.com/PennyLaneAI/pennylane-lightning/pull/416

### CODE REVIEW

**Review by AmintorDusko — APPROVED** [review]
Thank you @mlxd!
Please remember to add your name to the changelog.

**Review by multiphaseCFD — APPROVED** [review]
Thanks @mlxd for the nice work!

File: pennylane_lightning_kokkos/lightning_kokkos.py [hunk 1]
```diff
@@ -276,6 +276,11 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
         Returns:
             array[float]: list of the probabilities
         """
+        if wires and len(wires) > 1 and (not np.all(list(wires)[:-1] <= list(wires)[1:])):
```
Comment (timmysilv) [inline pennylane_lightning_kokkos/lightning_kokkos.py:279]: this check should really be done with `device_wires` below after line 291. Example of why:
```pycon
>>> dev = qml.device('lightning.qubit', wires=[2, 0, 1])
>>> obs = qml.PauliZ(2) @ qml.Identity(0) @ qml.Identity(1)
>>> dev.map_wires(obs.wires)  # should be accepted
<Wires = [0, 1, 2]>
>>> obs2 = qml.PauliZ(0) @ qml.Identity(1) @ qml.Identity(2)
>>> dev.map_wires(obs2.wires)
<Wires = [1, 2, 0]>  # should be rejected
```

Comment (mlxd) [reply pennylane_lightning_kokkos/lightning_kokkos.py:279]: Yep, good call. It's probably worth adding the above test case to the pennylane `pl-device-tests` test runners, to ensure we catch such things early on external devices.

Comment (mlxd) [reply pennylane_lightning_kokkos/lightning_kokkos.py:279]: Also, we will be be releasing with this forbidden behaviour now to allow time for a follow-up fix later in the 0.29 release cycle.

**Pull request merged**